Realm/RLMArrayLinkView.mm

```diff
@@ -79,6 +79,10 @@ - (NSUInteger)count {
     return _backingLinkView->size();
 }
 
+- (BOOL)isInvalidated {
+    return !_backingLinkView->is_attached() || !RLMMatchesThread(_realm);
```

Comment:
I started writing my reasoning and ended up agreeing with you in the process :smile:. I removed that check.